auth/env_keychain.go

```diff
@@ -12,27 +12,37 @@ import (
 	"github.com/pkg/errors"
 )
 
-// NewKeychain returns either a EnvKeychain or a authn.DefaultKeychain depending on whether the provided environment variable is set
-func NewKeychain(envVar string) authn.Keychain {
+// ResolveKeychain returns either:
+// * a resolved keychain from the ggcr DefaultKeychain or
+// * a multi-keychain with a resolved keychain from the provided environment variable, and a resolved keychain from the ggcr DefaultKeychain
+// depending on whether the provided environment variable is set
```

Comment:
1) I previously used `*` bullet points in doc comments, but I recently realized it looks _terrible_ https://godoc.org/github.com/buildpacks/lifecycle/layers#Factory.SliceLayers.

2) I found this comment very hard to understand the first time I read it. I think if we leave out some of the if/then logic here it is easier to understand. Whether or not the env keychain is technically included in the multi keychain doesn't matter to the consumer? It only matter that if the env var is set that credential source takes precedence.

Maybe something like the following?
```suggestion
// ResolveKeychain returns a keychain containing authentication configuration for the given images from the following sources, if they exist, in order of precedence: the provided environment variable and the docker config.json file.
```